addons/sale_loyalty/tests/test_program_numbers.py

```diff
@@ -1958,3 +1958,55 @@ def test_apply_order_and_specific_discounts(self):
             order.amount_total,
             "Order total should be 0, as a specific discount should have been applied.",
         )
+
+    def test_100_percent_discount_rounding(self):
+        self.env['loyalty.program'].create({
+            'name': '100% Discount',
+            'program_type': 'coupons',
+            'trigger': 'with_code',
+            'applies_on': 'current',
+            'rule_ids': [(0, 0, {
+                'code': '100pc',
+                'mode': 'with_code',
+            })],
+            'reward_ids': [(0, 0, {
+                'reward_type': 'discount',
+                'discount': 100,
+                'discount_mode': 'percent',
+                'discount_applicability': 'order',
+            })],
+        })
+
+        tax_9_99pc = self.env['account.tax'].create({
+            'name': "19.99% Tax",
+            'amount_type': 'percent',
+            'amount': 19.99,
+        })
+
+        prices = [19.99, 19.99, 19.99]
+
+        product = self.env['product.product'].create({
+            'name': 'Test Product',
+            'list_price': 100,
+            'taxes_id': [(6, 0, [tax_9_99pc.id])],
+        })
+
+        order_lines = []
+        for price in prices:
+            order_lines.append((0, 0, {
+                'product_id': product.id,
+                'product_uom_qty': 1,
+                'price_unit': price,
+                'tax_id': [(6, 0, [tax_9_99pc.id])],
+            }))
+
+        order = self.env['sale.order'].create({
+            'partner_id': self.env['res.partner'].create({'name': 'Test Partner'}).id,
+            'order_line': order_lines
+        })
```

Comment:
As you know you want to test with 2 SOL having the same product, you may want to simplify this part of the test with something like
```suggestion
        order = self.empty_order
        line_vals = {
            'product_id': self.product_A.id,
            'product_uom_qty': 1,
            'price_unit': 9.99,
            'tax_id': [(6, 0, [tax_19_99pc.id])],
        }
        self.empty_order.write({
            'order_line': [Command.create(line_vals), Command.create(line_vals)],
        })
```